src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -84,36 +139,89 @@ private async Task InvokeInternal(HttpContext context)
 
                 if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestProtocol))
                 {
-                    AddToList(list, nameof(request.Protocol), request.Protocol);
+                    if (httpEnabled)
+                    {
+                        AddToList(list, nameof(request.Protocol), request.Protocol);
+                    }
+                    if (w3cEnabled)
+                    {
+                        AddToList(w3cList, nameof(request.Protocol), request.Protocol);
+                    }
                 }
 
                 if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestMethod))
                 {
-                    AddToList(list, nameof(request.Method), request.Method);
+                    if (httpEnabled)
+                    {
+                        AddToList(list, nameof(request.Method), request.Method);
+                    }
+                    if (w3cEnabled)
+                    {
+                        AddToList(w3cList, nameof(request.Method), request.Method);
+                    }
                 }
 
-                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestScheme))
+                if (httpEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestScheme))
                 {
                     AddToList(list, nameof(request.Scheme), request.Scheme);
                 }
 
-                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestPath))
+                if (httpEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestPath))
                 {
                     AddToList(list, nameof(request.PathBase), request.PathBase);
                     AddToList(list, nameof(request.Path), request.Path);
                 }
 
-                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestQuery))
+                if (httpEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestQueryString))
                 {
                     AddToList(list, nameof(request.QueryString), request.QueryString.Value);
                 }
 
+                if (w3cEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestQuery))
```

Comment:
Why is this separate from the query string?

Reply:
Good question, it doesn't need to be. Fixed: https://github.com/dotnet/aspnetcore/pull/33251/commits/492ab9821c94322498c299a9fb406702bf8b5a96